market_analysis/views/default.py

```diff
@@ -0,0 +1,450 @@
+                                    total_shares, round(total_value, 2)
+                                    )
+
+        export['stocks'] = stocks
+        export['total_shares'] = total_shares
+        export['total_value'] = round((total_value), 2)
+
+        return {'entry': export}
+
+    else:
+        return HTTPFound(request.route_url('api_error'))
+
+
+@view_config(route_name='new_user', renderer='templates/new_user.jinja2')
+def new_user(request):
+    """New User form."""
+    username = password = password_verify = first_name = ''
+    last_name = phone_number = email = error = message = ''
+
+    if request.method == 'POST':
+        username = request.POST['username']
+        password = request.POST['password']
+        password_verify = request.POST['password_verify']
+        first_name = request.POST['first_name']
+        last_name = request.POST['last_name']
+        phone_number = request.POST['phone_number']
+        email = request.POST['email']
+
+        try:
+            query = request.dbsession.query(Users)
+            result = query.filter_by(username=username).first()
+        except DBAPIError:  # pragma: no cover
+            return Response(db_err_msg, content_type='text/plain', status=500)
+
+        if result:
+            message = 'User "{}" already exists.'.format(username)
+        else:
+            if username != '' and password != '' and password_verify != ''\
+               and first_name != '' and last_name != '' and email != '':
+
+                if (password == password_verify) and (len(password) > 6):
+                    message = "good job, you can enter info"
+                    date_joined = datetime.datetime.now()
+                    date_last_logged = datetime.datetime.now()
+                    new = Users(
+                        username=username,
+                        first_name=first_name,
+                        last_name=last_name,
+                        email=email,
+                        email_verified=False,
+                        date_joined=date_joined,
+                        date_last_logged=date_last_logged,
+                        pass_hash=pwd_context.encrypt(password),
+                        phone_number=phone_number,
+                        phone_number_verified=False,
+                        active=True,
+                        password_last_changed=datetime.datetime.now(),
+                        password_expired=False,
+                        is_admin=0,
+                    )
+                    request.dbsession.add(new)
+                    headers = remember(request, username)
+                    return HTTPFound(location=request.route_url('search'),
+                                     headers=headers)
+                else:
+                    error = 'Passwords do not match or password'\
+                            ' is less than 6 characters'
+            else:
+                error = 'Missing Required Fields'
```

Comment:
The benefit is that you can test the validation of your user data and the validation of the supplied passwords without needing to engage any systems like the database.